src/modules/Dashboard/components/DashboardTable/DashboardTable.tsx

```diff
@@ -8,39 +8,40 @@ import { StyledEllipsisText } from 'shared/styles';
 import { DashboardTableProps } from './DashboardTable.types';
 import { StyledTableCell, StyledTableHead } from './DashboardTable.styles';
 
-export const DashboardTable = ({
-  columns,
-  rows,
-  keyExtractor = (_row: Row, index: number) => `row-${index}`,
-  maxHeight = '100%',
-  uiType = UiType.Primary,
-  className = '',
-  order,
-  orderBy,
-  emptyComponent,
-  page,
-  handleRequestSort,
-  handleChangePage,
-  count,
-  hasColFixedWidth,
-  rowsPerPage = DEFAULT_ROWS_PER_PAGE,
-  onScroll,
-  'data-testid': dataTestid,
-}: DashboardTableProps) => {
-  const tableHeader = (
-    <StyledTableCellContent uiType={uiType}>
-      <TablePagination
-        component="div"
-        count={count}
-        rowsPerPage={rowsPerPage}
-        page={page - 1}
-        onPageChange={handleChangePage}
-        labelRowsPerPage=""
-        rowsPerPageOptions={[]}
-        data-testid={`${dataTestid}-table-pagination`}
-      />
-    </StyledTableCellContent>
-  );
+export const DashboardTable = (props: DashboardTableProps) => {
```

Comment:
Since our `props` are a discriminated union, I couldn't restructure it directly in the parameter, because I needed to access `props.enablePagination` as written a few lines later, in order to prevent our type-checker from losing track of which of the union's component types we want to narrow down to.